assets/javascripts/discourse/models/discourse-post-event-event.js

```diff
@@ -1,70 +1,170 @@
+    this._stats = this.#initStatsModel(stats);
+  }
+
+  get reminders() {
+    return this._reminders;
+  }
 
-const STATUSES = {
-  standalone: 0,
-  public: 1,
-  private: 2,
+  set reminders(reminders = []) {
+    this._reminders = new TrackedArray(reminders);
+  }
+
+  get creator() {
+    return this._creator;
+  }
+
+  set creator(user) {
+    this._creator = this.#initUserModel(user);
+  }
+
+  get isPublic() {
+    return this.status === "public";
+  }
+
+  get isPrivate() {
+    return this.status === "private";
+  }
+
+  updateFromEvent(event) {
+    this.name = event.name;
+    this.startsAt = event.startsAt;
+    this.endsAt = event.endsAt;
+    this.url = event.url;
+    this.timezone = event.timezone;
+    this.status = event.status;
+    this.creator = event.creator;
+    this.isClosed = event.isClosed;
+    this.isExpired = event.isExpired;
+    this.isStandalone = event.isStandalone;
+    this.minimal = event.minimal;
+    this.recurrenceRule = event.recurrenceRule;
+    this.canUpdateAttendance = event.canUpdateAttendance;
+    this.canActOnDiscoursePostEvent = event.canActOnDiscoursePostEvent;
+    this.shouldDisplayInvitees = event.shouldDisplayInvitees;
+    this.stats = event.stats;
+    this.sampleInvitees = event.sampleInvitees;
+    this.reminders = event.reminders;
+  }
+
+  #initUserModel(user) {
+    if (!user || user instanceof User) {
+      return user;
+    }
+
+    return User.create(user);
+  }
+
+  #initStatsModel(stats) {
+    if (!stats || stats instanceof DiscoursePostEventEventStats) {
+      return stats;
+    }
+
+    return DiscoursePostEventEventStats.create(stats);
+  }
+}
+
+// TODO (joffrey): get rid of all the following
+const DEFAULT_REMINDER = {
+  type: "notification",
+  value: 15,
+  unit: "minutes",
+  period: "before",
 };
 
-const Event = RestModel.extend({
-  init() {
-    this._super(...arguments);
-
-    this.__type = "discourse-post-event-event";
-  },
-
-  update(data) {
-    return ajax(`/discourse-post-event/events/${this.id}.json`, {
-      type: "PUT",
-      dataType: "json",
-      contentType: "application/json",
-      data: JSON.stringify({ event: data }),
-    });
-  },
-
-  updateProperties() {
-    const attributesKeys = Object.keys(ATTRIBUTES);
-    return this.getProperties(attributesKeys);
-  },
-
-  createProperties() {
-    const attributesKeys = Object.keys(ATTRIBUTES);
-    return this.getProperties(attributesKeys);
-  },
-
-  _transformProps(props) {
-    const attributesKeys = Object.keys(ATTRIBUTES);
-    attributesKeys.forEach((key) => {
-      const attribute = ATTRIBUTES[key];
-      if (attribute?.transform) {
-        props[key] = attribute.transform(props[key]);
-      }
-    });
-  },
-
-  beforeUpdate(props) {
-    this._transformProps(props);
-  },
-
-  beforeCreate(props) {
-    this._transformProps(props);
-  },
-});
-
-export default Event;
+export function removeReminder(event, reminder) {
+  return event.reminders.removeObject(reminder);
```

Comment:
I think we should avoid using ember's removeObject here?

Reply:
yes I wanted to keep this for later but let's do it